app/controllers/user/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,28 @@
+class User::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def facebook
+    if user_signed_in?
+      link_current_account_to_facebook
+    else
+      sign_in_with_facebook
+    end
+  end
+
+  private
+
+  def link_current_account_to_facebook
+    # TODO: Implement
+  end
+
+  def sign_in_with_facebook
+    auth = request.env['omniauth.auth']
+    @user = User.from_omniauth(auth)
+    if @user.persisted?
+      sign_in_and_redirect(@user, event: :authentication)
+      set_flash_message(:notice, :success,
+                        kind: t('user.omniauth_callbacks.facebook.kind')) if is_navigational_format?
```

Comment:
I guess the same reason here:
https://github.com/Coursemology/coursemology2/blob/master/app/controllers/course/user_registrations_controller.rb#L50